### PR TITLE
Show non-US states(Other) in Committee Schedule A by state aggregates

### PR DESCRIPTION
**Pull request by fec-jli**
## Summary
Example link:
[https://www.fec.gov/data/committee/C00370007/?tab=raising](https://www.fec.gov/data/committee/C00370007/?tab=raising)
on Individual contributions section in committee profile page, click Group by State, we will group by all non-US state values into one new row called: `Other`

- Resolves # [https://github.com/fecgov/openFEC/issues/3246](https://github.com/fecgov/openFEC/issues/3246)

related fec-cms Issue: [https://github.com/fecgov/fec-cms/issues/2237](https://github.com/fecgov/fec-cms/issues/2237)

_Include a summary of proposed changes._
tag: receipts
Path:/committee/{committee_id}/schedules/schedule_a/by_state/
endpoint: resources---aggregates.ScheduleAByStateView
model. aggregates.py—ScheduleAByState
new view: public.ofec_sched_a_agg_state_vw
table: disclosure.dsc_sched_a_aggregate_state

## How to test the changes locally
1)checkout branch: fix-sched_a-by-state-display to local openFEC
2) SQLA_CONN =dev db
3)start server
4)setup fec-cms point to your local api
5)test example: 
 [http://127.0.0.1:8000/data/committee/C00370007/?tab=raising](http://127.0.0.1:8000/data/committee/C00370007/?tab=raising)
6)also test sort by state and by total columns
screen shot 2018-08-01 at 11 09 45 am

### CODE REVIEW

File: data/migrations/V0098__add_ofec_sched_a_agg_state_vw.sql [hunk 1]
```diff
@@ -0,0 +1,32 @@
+DROP VIEW IF EXISTS public.ofec_sched_a_agg_state_vw;
+
+CREATE OR REPLACE VIEW public.ofec_sched_a_agg_state_vw AS
+select cmte_id,
```
Comment (vrajmohan) [inline data/migrations/V0098__add_ofec_sched_a_agg_state_vw.sql]: Can we make the layout easier to read and more consistent? I don't know if we ever completed the work @hcaofec was doing on database code conventions. Let's start with uppercase keywords, lowercase for all else, indented SELECT list, indented JOIN conditions, etc.

File: data/migrations/V0098__add_ofec_sched_a_agg_state_vw.sql [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ON sa.state = st.st
+GROUP BY CMTE_ID, CYCLE, st.st, STATE_FULL;
+
+-- grants
+ALTER TABLE public.ofec_sched_a_agg_state_vw OWNER TO fec;
+GRANT ALL ON TABLE public.ofec_sched_a_agg_state_vw TO fec;
+GRANT SELECT ON TABLE public.ofec_sched_a_agg_state_vw TO fec_read;
+
+
+/*
```
Comment (vrajmohan) [inline data/migrations/V0098__add_ofec_sched_a_agg_state_vw.sql]: Can we remove these lines?

File: tests/test_aggregates.py [hunk 3]
```diff
@@ -10,6 +10,7 @@
     ScheduleEByCandidateView,
     CommunicationCostByCandidateView,
     ElectioneeringByCandidateView,
+    ScheduleAByStateView,
```
Comment (vrajmohan) [inline tests/test_aggregates.py]: It would be easier for reading and maintenance if these were sorted.

File: tests/test_aggregates.py [hunk 4]
```diff
@@ -463,7 +513,7 @@ def test_by_size(self):
         }
         self.assertEqual(results[0], expected)
 
-    def test_by_state(self):
+    def test_by_state_candidate(self):
```
Comment (vrajmohan) [inline tests/test_aggregates.py]: Was this rename necessary?

File: webservices/common/models/aggregates.py [hunk 5]
```diff
@@ -20,40 +20,39 @@ class ScheduleABySize(BaseAggregate):
...
-    __table_args__ = {'schema' : 'disclosure'}
-    __tablename__ = 'dsc_sched_a_aggregate_state'
+    __tablename__ = 'ofec_sched_a_agg_state_vw'
     state = db.Column(db.String, primary_key=True, doc=docs.STATE_GENERIC)
     state_full = db.Column(db.String, primary_key=True, doc=docs.STATE_GENERIC)
 
 
 class ScheduleAByZip(BaseAggregate):
-    __table_args__ = {'schema' : 'disclosure'}
+    __table_args__ = {'schema': 'disclosure'}
```
Comment (vrajmohan) [inline webservices/common/models/aggregates.py]: Nice work on cleaning these up!

File: tests/test_aggregates.py [hunk 6]
```diff
@@ -35,6 +35,57 @@ def test_stable_sort(self):
...
+            ),
+        ]
+        results = self._results(
+            api.url_for(
+                ScheduleAByStateView,
+                committee_id='C0001',
+                cycle=2012
+            )
+        )
+        self.assertEqual(len(results), 1)
```
Comment (vrajmohan) [inline tests/test_aggregates.py]: This test module appears to use both `assert` and `self.assertEqual` with the former being more common. This would be a good opportunity to use `assert` everywhere. E.g. line 36.

File: tests/test_aggregates.py (hunk 3 above)
Comment (fec-jli) [reply tests/test_aggregates.py]: good suggestion.

**Comment by vrajmohan** [comment]
Sorry, @fec-jli. My recommendation was to use `assert` everywhere, not `self.assertEqual`.

**Comment by fec-jli** [comment]
@vrajmohan  Sorry, I changed back to assert. Thanks

File: data/migrations/V0100__add_ofec_sched_a_agg_state_vw.sql [hunk 7]
```diff
@@ -0,0 +1,19 @@
+DROP VIEW IF EXISTS public.ofec_sched_a_agg_state_vw;
+
+CREATE OR REPLACE VIEW public.ofec_sched_a_agg_state_vw AS
```
Comment (vrajmohan) [inline data/migrations/V0100__add_ofec_sched_a_agg_state_vw.sql]: Did you miss my earlier comment?
> Can we make the layout easier to read and more consistent? I don't know if we ever completed the work @hcaofec was doing on database code conventions. Let's start with uppercase keywords, lowercase for all else, indented SELECT list, indented JOIN conditions, etc.

Comment (fec-jli) [reply data/migrations/V0100__add_ofec_sched_a_agg_state_vw.sql]: Sorry, I didn't see your previous comment.
 good point, I will check the database code conventions with @hcaofec @fecjjeng . Thanks
